rpc/src/rpc.rs

```diff
@@ -4716,7 +4815,19 @@ pub mod tests {
 
     impl RpcHandler {
         fn start() -> Self {
-            let (bank_forks, mint_keypair, leader_vote_keypair) = new_bank_forks();
+            Self::_start(false)
+        }
+
+        fn start_with_secondary_indexes() -> Self {
+            Self::_start(true)
+        }
+
+        fn _start(enable_secondary_indexes: bool) -> Self {
```

Comment:
I don't love this approach.
Could we add a `RpcHandler::start_with_config(config: JsonRpcConfig)` method (called by `RpcHandler::start()` instead? Then just compose the JsonRpcConfig with whatever secondary indexes you want activated in the test.
Seems like control over those config fields might be useful in the future (or in some of the tests not using `RpcHandler::start()` anyway.

Reply:
Sure, I didn't love this either tbh. I agree, a config struct is more useful.